airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/task/DestinationTaskScopeProvider.kt

```diff
@@ -57,42 +51,49 @@ class DestinationTaskScopeProvider(config: DestinationConfiguration) :
     private val timeoutMs = config.gracefulCancellationTimeoutMs
 
     data class ControlScope(
-        val job: Job,
+        val name: String,
+        val job: CompletableJob,
         val dispatcher: CoroutineDispatcher,
-        val scope: CoroutineScope = CoroutineScope(dispatcher + job)
+        val scope: CoroutineScope = CoroutineScope(dispatcher + job),
+        val runningJobs: AtomicLong = AtomicLong(0)
```

Comment:
nit: does this need to be in the constructor, or can we just declare it as a private instance val?